solace/src/main/java/org/apache/apex/malhar/solace/AbstractSolaceBaseInputOperator.java

```diff
@@ -0,0 +1,319 @@
+  {
+    // TODO Auto-generated method stub
+
+  }
+
+  @Override
+  public void committed(long window)
+  {
+    try {
+      idempotentStorageManager.deleteUpTo(operatorId, window);
+    } catch (IOException e) {
+      DTThrowable.rethrow(e);
+    }
+
+  }
+
+  protected T processMessage(BytesXMLMessage message)
+  {
+    T tuple = convert(message);
+    if (tuple != null) {
+      emitTuple(tuple);
+    }
+    return tuple;
+  }
+
+
+  @Override
+  public void activate(Context.OperatorContext context)
+  {
+    try {
+      session.connect();
+      reliableConsumer = session.getMessageConsumer(rcHandler, cbHandler);
+      //consumer = getConsumer();
+      reliableConsumer.start();
+
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      consumer.stop();
+      clearConsumer();
+      consumer.close();
+      reliableConsumer.close();
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+  @Override
+  public void teardown()
+  {
+
+    idempotentStorageManager.teardown();
+    session.closeSession();
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    this.windowId = windowId;
+  }
+
+
+  protected abstract T convert(BytesXMLMessage message);
+
+  protected abstract void emitTuple(T tuple);
+
+  protected abstract Consumer getConsumer() throws JCSMPException;
+
+  protected abstract void clearConsumer() throws JCSMPException;
+
+  public void setProperties(JCSMPProperties properties)
+  {
+    this.properties = properties;
+
+  }
+
+
+  public IdempotentStorageManager getIdempotentStorageManager()
+  {
+    return idempotentStorageManager;
+  }
+
+  public void setUnackedMessageLimit(String unackedMessageLimit)
+  {
+    this.unackedMessageLimit = unackedMessageLimit;
+  }
+
+  public String getUnackedMessageLimit()
+  {
+    return unackedMessageLimit;
+  }
+
+  public void setConnectRetries(String connectRetries)
+  {
+    this.connectRetries = connectRetries;
+    logger.info("+++++++++++++++++++reconnectRetries: {}", this.connectRetries);
+  }
+
+  public void setReconnectRetries(String reconnectRetries)
+  {
+    this.reconnectRetries = reconnectRetries;
+    logger.info("+++++++++++++++++++reconnectRetries: {}", this.reconnectRetries);
+  }
+
+  public void setReapplySubscriptions(boolean state)
+  {
+    this.properties.setBooleanProperty(JCSMPProperties.REAPPLY_SUBSCRIPTIONS, state);
+  }
+
+  public void startConsumer()
+  {
+    try {
+      consumer = getConsumer();
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+
+  public class ReconnectCallbackHandler implements JCSMPReconnectEventHandler
+  {
+
+
+    @Override
+    public void postReconnect() throws JCSMPException
+    {
+
+      logger.info("++++++++++++++Solace client now Reconnected --  possibe Solace HA or DR fail-over +++++++++++++");
```

Comment:
The leading and trailing "++++" in the message is not the standard format in other log messages in malhar. We should standardize on log messages